packages/engine-core/src/binary/BinaryEngine.ts

```diff
@@ -236,36 +234,40 @@ You may have to run ${chalk.greenBright('prisma generate')} for your changes to
     this.checkForTooManyEngines()
   }
 
-  private setError(err: Error | RustLog | RustError) {
+  /**
+   * @deprecated do not rely on this
+   */
+  private setError(err: any) {
     if (isRustError(err)) {
-      this.lastRustError = err
-      this.logEmitter.emit(
-        'error',
-        new PrismaClientRustError({
-          clientVersion: this.clientVersion!,
-          error: err,
-        }),
-      )
-      if (err.is_panic) {
-        this.handlePanic()
-      }
+      this.handleRustError(err)
     } else if (isRustErrorLog(err)) {
       this.lastErrorLog = err
-      this.logEmitter.emit(
-        'error',
-        new PrismaClientRustError({
-          clientVersion: this.clientVersion!,
-          log: err,
-        }),
-      )
-      if (err.fields?.message === 'PANIC') {
-        this.handlePanic()
-      }
+      this.handleRustError(err)
     } else {
-      this.logEmitter.emit('error', err)
+      this.emitError(err)
+    }
+  }
+
+  private handleRustError(err: RustLog | RustError) {
+    const wrappedError = new PrismaClientRustError({
+      clientVersion: this.clientVersion!,
+      error: err,
+    })
+    this.emitError(wrappedError)
+    if (wrappedError.isPanic()) {
+      this.handlePanic()
     }
   }
 
+  private emitError(err: any) {
+    const message = err.message ?? err
+    this.logEmitter.emit('error', {
+      message: message,
+      timestamp: new Date(),
+      target: 'binary engine lifecycle management',
```

Comment:
Is this intentional?

Reply:
I made some changes based on https://github.com/prisma/prisma/pull/16287#discussion_r1029925661. This is done differently now.